api/rest/restapi.go

```diff
@@ -384,8 +385,31 @@ func (api *API) addFileHandler(w http.ResponseWriter, r *http.Request) {
 		sendAcceptedResponse(w, errors.New("unsupported media type"))
 		return
 	}
+	outChan := make(chan *ipld.Node)
+	ctx, cancel := context.WithCancel(context.Background())
+	go func() {
+		for nodePtr := range outChan {
+			node := *nodePtr
+			/* Send block data to ipfs */
+			pinS := types.PinSerial{}
+			err := api.rpcClient.Call("",
+				"Cluster",
+				"IPFSBlockPut",
+				node.RawData(),
+				&pinS)
 
-	err := dex.ImportToPrint(f)
+			/* Verify that block put cid matches*/
+			if node.String() != pinS.Cid { // node string is just cid string
+				logger.Warningf("mismatch. node cid: %s\nrpc cid: %s", node.String(), pinS.Cid)
+			}
+			if err != nil { // error will be triggered in importer
+				logger.Error(err)
+				cancel()
+				return
+			}
+		}
+	}()
+	err := dex.ImportToChannel(f, outChan, ctx)
```

Comment:
maybe I greenlighted this somewhere, but `ImportToChannel` has a very weird signature. It seems more natural like:

`func ImportToChannel(ctx, f) <-chan` (and the returned channel should be closed if context is cancelled).

Reply:
This is handled in PR #318, I'm not fixing it here too because there is not point modifying the dex repo and gx publishing now that PR #318 has brought this code to cluster.  I'll rebase and resolve conflicts after PR #318 is merged.